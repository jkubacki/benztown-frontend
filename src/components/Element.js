import React from 'react';
import { elementPropType } from 'constants/propTypes';
import ReactAudioPlayer from 'react-audio-player';

function Element({ element }) {
  return (
    <tr>
      <td>
        {element.attributes.name}
      </td>
      <td>
        <ReactAudioPlayer
          src={element.attributes.file}
          controls
        />
      </td>
      <td>
        { element.attributes.tag_list.join(', ') }
      </td>
    </tr>
  );
}

Element.propTypes = {
  element: elementPropType.isRequired,
};

export default Element;
