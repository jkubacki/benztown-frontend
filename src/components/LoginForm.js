import React from 'react';
import { compose } from 'redux'
import { connect } from 'react-redux'
import { Field, reduxForm } from 'redux-form';
import { login } from 'actions/auth';
import validateField from 'utils/validateField'

class LoginForm extends React.Component {
  handleSubmit = values =>
    this.props.login({
      username: values.email,
      password: values.password,
      rememberMe: values.remember,
    });

  render() {
    const { handleSubmit } = this.props;

    return(
      <form onSubmit={handleSubmit(this.handleSubmit)}>
        <div className="form-group">
          <label>Email address</label>
          <Field
            label="email"
            placeholder="Enter email"
            name="email"
            type="email"
            component={validateField}
            className="form-control"
            required={true}
          />
        </div>
        <div className="form-group">
          <label>Password</label>
          <Field
            label="password"
            placeholder="Enter password"
            name="password"
            type="password"
            component={validateField}
            className="form-control"
            required={true}
          />
        </div>
        <div className="form-group form-check">
          <Field
            label="Remember me"
            type="checkbox"
            name="remember"
            component="input"
            className="form-check-input"
          />
          <label className="form-check-label">Remember me</label>
        </div>
        <button type="submit" className="btn btn-outline-primary">Submit</button>
      </form>
    )
  }
}

export default compose(
  connect(
    null,
    { login },
  ),
  reduxForm({
    form: 'login',
    initialValues: {
      remember: true,
    },
  }),
)(LoginForm)
