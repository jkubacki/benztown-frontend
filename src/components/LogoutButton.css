@media all and (max-width:767px) {
  .btn-block-sm { width: 100%; display: block; }
}
