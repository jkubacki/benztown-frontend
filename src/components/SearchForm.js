import React from 'react';
import PropTypes from 'prop-types';
import { compose } from 'redux';
import { connect } from 'react-redux';
import getElementsAction from 'actions/elements';

class SearchForm extends React.Component {
  handleChange = (event) => {
    const searchQuery = event.target.value;
    const { getElements } = this.props;
    getElements({ q: searchQuery });
  };

  render() {
    return (
      <input type="text" placeholder="Search" className="form-control" onChange={this.handleChange} />
    );
  }
}

SearchForm.propTypes = {
  getElements: PropTypes.func.isRequired,
};

export { SearchForm as SearchFormUnwrapped };

export default compose(
  connect(
    null,
    { getElements: getElementsAction },
  ),
)(SearchForm);
