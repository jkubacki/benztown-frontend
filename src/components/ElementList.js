import React from 'react';
import PropTypes from 'prop-types';
import getElementsAction from 'actions/elements';
import { connect } from 'react-redux';

import Element from 'components/Element';

class ElementList extends React.Component {
  componentWillMount() {
    const { getElements } = this.props;
    getElements({});
  }

  render() {
    const { elements } = this.props;
    return (
      <table className="table">
        <thead>
          <tr>
            <th scope="col">Name</th>
            <th scope="col">File</th>
            <th scope="col">Tags</th>
          </tr>
        </thead>
        <tbody>
          {elements.map(element => <Element key={element.id} element={element} />)}
        </tbody>
      </table>
    );
  }
}

ElementList.propTypes = {
  elements: PropTypes.arrayOf(
    PropTypes.shape({
      attributes: PropTypes.shape({
        name: PropTypes.string.isRequired,
        file: PropTypes.string.isRequired,
        tag_list: PropTypes.array.isRequired,
      }).isRequired,
    }).isRequired,
  ).isRequired,
  getElements: PropTypes.func.isRequired,
};

function mapStateToProps(props) {
  const { elements } = props;
  return { elements };
}

export { ElementList }

export default connect(
  mapStateToProps,
  { getElements: getElementsAction },
)(ElementList);
