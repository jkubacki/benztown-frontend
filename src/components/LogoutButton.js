import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { compose } from 'redux';
import { logout as logoutAction } from 'actions/auth';

import './LogoutButton.css';

function LogoutButton({ logout }) {
  return (
    <button type="button" onClick={logout} className="btn btn btn-outline-dark btn-block-sm">
      Logout
    </button>
  );
}

LogoutButton.propTypes = {
  logout: PropTypes.func.isRequired,
};

export { LogoutButton as LogoutButtonUnwrapped };

export default compose(
  connect(
    null,
    { logout: logoutAction },
  ),
)(LogoutButton);
