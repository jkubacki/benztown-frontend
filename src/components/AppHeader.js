import React from 'react';
import { connect } from 'react-redux';
import { compose } from 'redux';

import LogoutButton from 'components/LogoutButton'
import SearchForm from 'components/SearchForm'

import ClientState from 'constants/clientState'

import './AppHeader.css';

function AppHeader({ isLoggedIn }) {
  return (
    <div className="navbar">
      <ul className="navbar-nav nav-fill d-flex w-100 flex-md-row">
        <li className="nav-item">
          <span className="navbar-brand">Benztown</span>
        </li>
        {
          isLoggedIn &&
          <li className="nav-item">
            <SearchForm />
          </li>
        }
        {
          isLoggedIn &&
          <li className="nav-item">
            <LogoutButton />
          </li>
        }
      </ul>
    </div>
  )
}

export default compose(
  connect(({ client }) => ({
    isLoggedIn: client.state === ClientState.LOGGED_IN,
  })),
)(AppHeader);
