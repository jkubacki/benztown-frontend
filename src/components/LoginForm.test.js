import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';

import { LoginForm } from 'components/LoginForm';

const props = {
  login: jest.fn(() => Promise.resolve()),
  handleSubmit: jest.fn(x => x),
};

describe('LoginForm', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<LoginForm {...props} />);
  });

  it('should render corectly', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render action button', () => {
    expect(wrapper.find('button').length).toEqual(1);
  });

  it('should render form', () => {
    expect(wrapper.find('form').length).toEqual(1);
  });

  it('should render 3 Field Component', () => {
    expect(wrapper.find('Field').length).toEqual(3);
  });

  it('should render email', () => {
    expect(wrapper.find('Field[name="email"]').length).toEqual(1);
  });

  it('should render password', () => {
    expect(wrapper.find('Field[name="password"]').length).toEqual(1);
  });

  it('should render checkbox', () => {
    expect(wrapper.find('Field[name="remember"]').length).toEqual(1);
  });

  it('should call login with correct input values', () => {
    const instance = wrapper.instance();

    instance.handleSubmit({
      email: 'admin@netguru.pl',
      password: 'adminadmin',
      remember: true,
    });

    expect(props.login).toHaveBeenCalledWith({
      username: 'admin@netguru.pl',
      password: 'adminadmin',
      rememberMe: true,
    });
  });
});
