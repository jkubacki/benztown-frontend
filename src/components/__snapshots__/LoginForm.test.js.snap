// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm should render corectly 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="form-group"
  >
    <label
      htmlFor="email"
    >
      Email address
    </label>
    <Field
      className="form-control"
      component={[Function]}
      label="email"
      name="email"
      placeholder="Enter email"
      required={true}
      type="email"
    />
  </div>
  <div
    className="form-group"
  >
    <label
      htmlFor="password"
    >
      Password
    </label>
    <Field
      className="form-control"
      component={[Function]}
      label="password"
      name="password"
      placeholder="Enter password"
      required={true}
      type="password"
    />
  </div>
  <div
    className="form-group form-check"
  >
    <Field
      className="form-check-input"
      component="input"
      label="Remember me"
      name="remember"
      type="checkbox"
    />
    <label
      className="form-check-label"
      htmlFor="remember"
    >
      Remember me
    </label>
  </div>
  <button
    className="btn btn-outline-primary"
    type="submit"
  >
    Submit
  </button>
</form>
`;
